ai/providers/vertexai.py

```diff
@@ -0,0 +1,111 @@
+import google.api_core.exceptions
+
+from .base_provider import BaseAPIProvider
+import os
+import logging
+import vertexai.generative_models
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+class VertexAPI(BaseAPIProvider):
+    VERTEX_AI_PROVIDER = "VertexAI"
+    MODELS = {
+        "gemini-1.5-flash-001": {
+            "name": "Gemini 1.5 Flash 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-1.5-flash-002": {
+            "name": "Gemini 1.5 Flash 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-1.5-pro-002": {
+            "name": "Gemini 1.5 Pro 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-1.5-pro-001": {
+            "name": "Gemini 1.5 Pro 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-1.0-pro-002": {
+            "name": "Gemini 1.0 Pro 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-1.0-pro-001": {
+            "name": "Gemini 1.0 Pro 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-flash-experimental": {
+            "name": "Gemini Flash Experimental",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-pro-experimental": {
+            "name": "Gemini Pro Experimental",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+        "gemini-experimental": {
+            "name": "Gemini Pro Experimental",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+        },
+    }
+
+    def __init__(self):
+        self.enabled = os.environ.get("VERTEX_AI_ENABLED", "").lower() in ["1", "true", "t", "yes", "y"]
```

Comment:
```suggestion
```

Related to the above comment, I think we can use the project ID as a status check 🔍